src/SurveyService.Data/GroupRepository.cs

```diff
@@ -26,5 +30,42 @@ public GroupRepository(IDataProvider provider)
     await _provider.SaveAsync();
     
     return dbGroup.Id;
+  }
+
+  public Task<DbGroup> GetAsync(GetGroupFilter filter)
+  {
+    if (filter is null)
+    {
+      return null;
+    }
+
+    IQueryable<DbGroup> dbGroup = _provider.Groups.AsQueryable();
+    dbGroup = dbGroup.Where(group => group.Id == filter.GroupId && group.IsActive);
+
+    if (filter.IncludeUserAnswers)
+    {

```

Comment:
group can't include answers and don't include questions
if (includequestions)
{
xxx (only active questions)
if (include options)
{ xxx
.........
}}
in lists of questions and options (without answers) there are only active
if includeanswers - > include all options

Reply:
For this reason, I check filter.IncludeUserAnswers first. If I check includeQuestions first, then "if(filter.includeUserAnswers)" will not execute

Reply:
if request is correct and user needs answers, there would be in filter:
includeQuestions = true, includeOptions = true and includeUserAnswers =true
so, you first check includeUserAnswers -> it's true -> you add questions, options and answers;
then you check includeOptions -> it's true too -> you add questions and options again;
then you check includeQuestions -> it's true too -> you add questions third time;

i think, you need first check includeQuestions: if it is true, check includeOptions (if inside if) and so on

Reply:
Done